jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/ServletPathSpec.java

```diff
@@ -270,6 +284,44 @@ private boolean isWildcardMatch(String path)
         return false;
     }
 
+    @Override
+    public MatchedPath matched(String path)
+    {
+        switch (_group)
+        {
+            case EXACT:
+                if (_declaration.equals(path))
+                    return MatchedPath.from(path, null);
+                break;
+            case PREFIX_GLOB:
+                if (isWildcardMatch(path))
+                {
+                    String pathMatch = path;
+                    String pathInfo = null;
+                    if (path.length() != (_specLength - 2))
+                    {
+                        pathMatch = path.substring(0, _specLength - 2);
+                        pathInfo = path.substring(_specLength - 2);
+                    }
+                    return MatchedPath.from(pathMatch, pathInfo);
```

Comment:
This also needs to avoid allocation in the common case of "/foo" matching "/foo/*"

```suggestion
                if (isWildcardMatch(path))
                {
                    if (path.length() == (_specLength - 2))
                        return finalFieldAllocatedInConstructor;
                    String pathMatch = anotherFinalFiledAllocatedInConstructor ;
                    String pathInfo = path.substring(_specLength - 2);
                    return MatchedPath.from(pathMatch, pathInfo);
```

Reply:
You'll have a map of values, mapping each PathSpec and potential path.

Reply:
The pathSpec itself should know it's prefix, so that is what is returned.  No map needed.